 # 0x05 - Python - Exceptions
Task # | Short Description
-------|------------
[Task 0](0-safe_print_list.py) | Prints `x` elements of a list
[Task 1](100-safe_print_integer_err.py) | Prints an integer
[Task 2](2-safe_print_list_integers.py) | Prints the first `x` elements of a list & only integers.
[Task 3](3-safe_print_division.py) | Divides 2 integers & prints the result.
[Task 4](4-list_division.py) | Divides element by element 2 lists.
[Task 5](5-raise_exception.py) | Raises a type exception.
[Task 6](6-raise_exception_msg.py) | Raises a name exception with a message. 
[Task 7](100-safe_print_integer_err.py) | **ADVANCED** Prints an integer
[Task 8](101-safe_function.py) | **ADVANCED** Executes a function safely.
[Task 9](102-magic_calculation.py) | Write the Python function `def magic_calculation(a, b):` that does exactly the same as an example of Pyton bytecode.
[Task 10](103-python.c) | **ADVANCED** Create three C functions that print some basic info about Python lists, Python bytes and Python float objects.

 ## Lessons Learned
* Why Python programming is awesome
* What's the difference between errors & exceptions
* What are exceptions & how to use them
* When do we need to use exceptions
* How to correctly handle an exception
* What's the purpose of catching exceptions
* How to raise a builtin exception
* When do we need to implement a clean-up action after an exception
